src/main/java/com/minecolonies/core/event/DataPackSyncEventHandler.java

```diff
@@ -121,6 +118,21 @@ else if (event.getPlayer().getGameProfile() != owner)
             }
         }
 
+        /**
+         * Sends compat manager data on login, has to be after datapack sync registries are loaded
+         *
+         * @param event
+         */
+        @SubscribeEvent(priority = EventPriority.LOWEST)
+        public static void sendOnLogin(final PlayerEvent.PlayerLoggedInEvent event)
```

Comment:
Is this always 100% triggered after datapack sync? Or could there be a race condition?

Reply:
PlayerLoggedIn is one of the last events to fire in the sequence.

I'm not sure I understand why DataPackSync was not sufficient, though.  That does trigger before recipes/tags (otherwise we wouldn't need to serialize as much stuff ourselves), but it should still be after registries are synched.  Unless it's a neoforge bug?

Reply:
it could be a neoforge bug not sure, but the resulting stacks during deserialize on client side are using the old registries which are getting replaced after the datapack load finishes and then error when the client tries to serialize them with the updated registries.
And no there is no race condition, the events are all on the server thread not run concurrently

Reply:
You might need to check what happens on `/reload` as well then.  The events get fired in different orders for that case.